File: 11_oop_abstrakčios klasės/Ziurkenas.java

public class Ziurkenas extends Gyvunas {

	private int uodegosIlgis;
	private String spalva;

	public Ziurkenas(String name) {
		super(name, "Ziurkenas");

	}

	@Override
	public String toString() {
		return "" + this.getType() + ". " + "Vardas: " + this.getName() + ", uodegos ilgis: " + this.uodegosIlgis + "cm, spalva: " + this.spalva;
	}

	public int getUodegosIlgis() {
		return this.uodegosIlgis;
	}

	public void setUodegosIlgis(int uodegosIlgis) {
		this.uodegosIlgis = uodegosIlgis;
	}

	public String getSpalva() {
		return this.spalva;
	}

	public void setSpalva(String spalva) {
		this.spalva = spalva;
	}
}

File: 11_oop_abstrakčios klasės/WindowsFormsApp10_gyvunai/WindowsFormsApp10_gyvunai/Ziurkenas.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WindowsFormsApp10_gyvunai
{
    internal class Ziurkenas : Gyvunas
    {
        private int uodegosIlgis;
        private String spalva;

        public int UodegosIlgis { get => uodegosIlgis; set => uodegosIlgis = value; }
        public string Spalva { get => spalva; set => spalva = value; }

        public Ziurkenas(String name) : base(name, "ziurkenas") {}

        public override string ToString()
        {
            return $"Gyvunas: {Type}, vardas: {Name}, uodegos ilgis: {uodegosIlgis}, " +
                $"spalva: {spalva}";
        }
    }
}
